File: app/src/main/java/io/sweers/catchup/rx/boundlifecycle/observers/adapter/SubscriberAdapter.java
package io.sweers.catchup.rx.boundlifecycle.observers.adapter;

import org.reactivestreams.Subscriber;
import org.reactivestreams.Subscription;

public abstract class SubscriberAdapter<T> implements Subscriber<T> {
  public SubscriberAdapter() {
    super();
  }

  @Override
  public final int hashCode() {
    return super.hashCode();
  }

  @Override
  public final boolean equals(Object obj) {
    return super.equals(obj);
  }

  @Override
  protected final Object clone() throws CloneNotSupportedException {
    return super.clone();
  }

  @Override
  public final String toString() {
    return super.toString();
  }

  @Override
  protected final void finalize() throws Throwable {
    super.finalize();
  }

  @Override
  public void onSubscribe(Subscription s) {

  }

  @Override
  public void onNext(T t) {

  }

  @Override
  public void onError(Throwable t) {

  }

  @Override
  public void onComplete() {

  }
}

File: app/src/main/kotlin/io/sweers/catchup/rx/observers/adapter/SubscriberAdapter.kt
/*
 * Copyright (c) 2017 Zac Sweers
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.sweers.catchup.rx.observers.adapter

import org.reactivestreams.Subscriber
import org.reactivestreams.Subscription

abstract class SubscriberAdapter<T> : Subscriber<T> {

  override fun hashCode(): Int {
    return super.hashCode()
  }

  override fun equals(other: Any?): Boolean {
    return super.equals(other)
  }

  override fun onSubscribe(s: Subscription) {

  }

  override fun onNext(t: T) {

  }

  override fun onError(t: Throwable) {

  }

  override fun onComplete() {

  }
}
